<launch>
    <!--############### Node Configurations ##################-->
    <arg name="output" default="screen"/>
    <arg name="sync_rate" default="20"/>
    <arg name="num_cameras" default="1"/>

    <!--############### IMU Configurations ###################-->
    <!-- Connection -->
    <arg name="port" default="/dev/serial/by-id/usb-FTDI_TTL232R-3V3_FT8Y6SOG-if00-port0"/>

    <!-- Frame ID for messages -->
    <arg name="imu_frame_id" default="imu"/>

    <!-- Baudrate of serial comms (see manual for allowed values) -->
    <arg name="baudrate" default="921600"/>

    <!-- Data rate in Hz -->
    <arg name="imu_rate" default="200"/>

    <!-- The unit of pulse width (us) -->
    <arg name="sync_pulse_width_us" default="1000"/>

    <!-- Ros Topic settings -->
    <arg name="enable_mag" default="true"/>
    <arg name="enable_pres" default="true"/>
    <arg name="enable_temp" default="true"/>

    <!-- Camera Configurations -->
    <!-- Common Interface -->
    <arg name="frame_rate" default="$(arg sync_rate)"/>
    <arg name="cam0" default="15322783"/>

    <group ns="sync/imu">
        <param name="port" type="string" value="$(arg port)"/>
        <param name="baudrate" type="int" value="$(arg baudrate)"/>
        <param name="imu_rate" type="int" value="$(arg imu_rate)"/>
        <param name="enable_mag" type="bool" value="$(arg enable_mag)"/>
        <param name="enable_pres" type="bool" value="$(arg enable_pres)"/>
        <param name="enable_temp" type="bool" value="$(arg enable_temp)"/>
        <param name="sync_rate" type="int" value="$(arg sync_rate)"/>
        <param name="sync_pulse_width_us" type="int" value="$(arg sync_pulse_width_us)"/>
    </group>

    <!-- Sync Node -->
    <node pkg="cam_imu_sync" type="cam_imu_sync_node" name="sync" output="$(arg output)">
        <param name="num_cameras" type="int" value="$(arg num_cameras)"/>
        <param name="fps" type="double" value="$(arg frame_rate)"/>

        <param name="cam0/identifier" type="string" value="$(arg cam0)"/>
        <param name="cam0/camera_name" type="string" value="pg_$(arg cam0)"/>
        <param name="cam0/calib_url" type="string" value="file://${ROS_HOME}/camera_info/pg_$(arg cam0).yaml"/>

        <param name="auto_shutter" type="bool" value="false"/>
        <param name="shutter_ms" type="double" value="6"/>
        <param name="auto_gain" type="bool" value="false"/>
        <param name="gain_db" type="double" value="0"/>
        <param name="trigger_source" type="int" value="2"/>
        <param name="trigger_polarity" type="int" value="1"/>
    </node>
</launch>
